### ISSUE TITLE
(Feature) All-In-One extension seed

### ISSUE THREAD

**Issue by xdevnull** [Thalimenel/Chrome-Extension-Seed#4]
**TLDR**
**Support multiple browsers**

Currently, The repo support Chrome/Opera _(chromium based)_ but i thought we could take the repo one step further by supporting all browsers. One code which runs in all browsers.

According to MDN

> The differences between browsers’ extensions API implementations fall into four areas: namespace, asynchronous event handling, API coverage, and manifest keys.

For the **namespace**, We could inject the namespace variable right before ```typescript compile``` commands run. In this case, we need to hook the compile command. 

Example:
```npm run pack --chrome``` -> replace namespace with **chrome**
```npm run pack --firefox``` -> replace namespace with **browser**

The other difference, **Asynchronous event handling**. Only firefox allows "Promises" other browsers only support callbacks. In this part, I thought about create class which serves as adapter and force the client to use "callbacks" as unified api.

API coverage: Don't know what to do, Edge for example does not support privacy function. But in case we create a wrapper and check whether a given function exists before calling.

**Manifest keys** we could remove unsupported keys before build. 

Not sure whether it's a good thing to do. Let me know what you think.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Differences_between_API_implementations

**Comment by Thalimenel** [comment]
lot's of stuff to discuss.

first the easier part:
Edge - newer edge is chrome based. you can use same extensions (and the google store) with it.
I wouldn't do anything specific for old Edge

re firefox and compatibility: my general approach is "use the right tool for the job". so I would just use a different seed for firefox, otherwise you will be chasing your tail going after support issues.

btw, unrelated, but currently, and for the foreseeable future, firefox supports the chrome namespace.
